src/api/user.ts

```diff
@@ -0,0 +1,31 @@
+import { apiPost, apiGet } from './base';
+import { User, ApiError, FormData } from '../types';
+import { ERROR_MESSAGE } from '../utils/constants';
+
+export const authUser = async (
+  formData: FormData,
+  onError?: (error: ApiError) => void,
+) => {
+  try {
+    return await apiPost('user/login', formData, undefined, onError);
+  } catch (error: unknown) {
+    const Error = error as any;
+    if (Error.status === 400) {
+      onError?.({
+        status: 400,
+        message: ERROR_MESSAGE.INVALID_CREDENTIALS,
+      });
+    }
+    throw Error;
+  }
+};
```

Comment:
I dont think this should be handled here. I don't think it needs any additional handling. You would be fine just with the code you have with the `setErrorMessage`. 

If you want to handle it here, cause you want to have standardized error messages, it should be done with something like 

```suggestion
  onError?: (error: ApiError) => void,
) => {
const handleError = (error) => {
    if (Error.status === 400) {
      onError?.({
        status: 400,
        message: ERROR_MESSAGE.INVALID_CREDENTIALS,
      });
    }
    }
    
    return error
    }

    return await apiPost('user/login', formData, undefined, handleError);
  
};
```

avoid the re-throwing.

Reply:
but! the important thing. I dont think this should be handled on frontend in this manner. You should just check what is the error status and use the appropriate localized (in our case just czech) message for the UI.